core/src/main/java/org/hibernate/ogm/dialect/spi/AssociationTypeContext.java

```diff
@@ -20,13 +20,20 @@
 public interface AssociationTypeContext {
 
 	/**
-	 * Provide access to the options set for an entity or property.
+	 * Provide access to the options set for the association.
 	 *
-	 * @return a context object providing access to the options effectively applying for a given entity or property.
+	 * @return a context object providing access to the options effectively applying for the association.
 	 */
 	OptionsContext getOptionsContext();
 
 	/**
+	 * Provide access to the options set for the entity owner of the association.
+	 *
+	 * @return a context object providing access to the options effectively applying for the owner entity.
+	 */
+	OptionsContext getOwnerEntityOptionsContext();
```

Comment:
> The fact is that the decision to use one or the other is taken in the
dialect depending on if the association is stored in the entity or not

Yes, understood. What I meant is, can we somehow generalize that, though? I
suppose it's tricky as "in-entity" vs. "separate" is not a concept in core,
IIRC.